gadget/update.go

```diff
@@ -1393,52 +1429,65 @@ func isSameRelativeOffset(one *RelativeOffset, two *RelativeOffset) bool {
 	return false
 }
 
-func isLegacyMBRTransition(from *LaidOutStructure, to *LaidOutStructure) bool {
+func isLegacyMBRTransition(from *VolumeStructure, to *VolumeStructure) bool {
 	// legacy MBR could have been specified by setting type: mbr, with no
 	// role
-	return from.Type() == schemaMBR && to.Role() == schemaMBR
+	return from.Type == schemaMBR && to.Role == schemaMBR
+}
+
+func sizeSegmentsIntersect(from *VolumeStructure, to *VolumeStructure) bool {
```

Comment:
shouldn't this be called segmentSizesAreCompatible?

Reply:
Changed